akka-http/client/src/main/scala/endpoints4s/akkahttp/client/Endpoints.scala

```diff
@@ -95,7 +96,7 @@ trait EndpointsWithCustomErrors
       case x                             => throw InvalidHeaderDefinition(x)
     }
 
-  type Request[A] = A => Future[HttpResponse]
+  type Request[A] = A => HttpRequest
```

Comment:
Would it be sensible to keep the Future in this type? i.e.:
```scala
  type Request[A] = A => Future[HttpRequest]
```
When we handle a request with authentication, the HTTP credentials are validated using a separate microservice. With the old type we could leverage the Future to perform that validation via a separate and get the necessary account data. 
I realize this is kind of an edge case, but it would make migration for us easier and since the Future was already there I can imagine that others may depend on this too.

I've pushed a working branch here, the change wasn't too impactful: https://github.com/simacan/endpoints4s/tree/middleware-take-two